ddtrace/context.py

```diff
@@ -94,6 +94,15 @@ def sampling_priority(self, value):
                 return
             self._metrics[SAMPLING_PRIORITY_KEY] = value
 
+    @property
+    def traceparent(self):
+        # type: () -> str
+        if self.trace_id is None or self.span_id is None:
+            return ""
+
+        sampled = 1 if self.sampling_priority and self.sampling_priority > 0 else 0
+        return "00-{:032x}-{:016x}-{:02x}".format(self.trace_id, self.span_id, sampled)
```

Comment:
Do we expect this property to be accessed multiple times for the same span? trace id? Are there any performance concerns with having an internal representation that needs to be converted during the lifecycle of a span?

Reply:
The following benchmark converts traceid, spanid, and sample rate to hex then generates a traceparent. This ran 1 million operations in 0.433 seconds (0.43 microseconds per string): 
```
import timeit
timeit.timeit(setup='from ddtrace import Span; s = Span("a");', stmt='"00-{:032x}-{:016x}-{:02x}".format(s.trace_id, s.span_id, 0)', number=1000000)
```

I updated the benchmark to only convert the sample rate to hex and then generate the traceparent. This ran in 0.245 seconds (0.245 microseconds per string): 
```
import timeit
timeit.timeit(setup='from ddtrace import Span; s = Span("a");', stmt='"00-00000000000000009ab1134a468e7936-7473f04b601bdbc0-{:02x}".format(0)', number=1000000)
```

Reply:
Yes, this is helpful!

I was wondering what benefit we gain from not repeatedly converting the `trace_id` for multiple spans.

For a trace with 100 db spans, constructing the `traceparent` header from the `trace_id` and `span_id` took on my machine 24.5us.

```
root@b1754db4c2d8:/# python -m timeit --number=100000 --setup='from ddtrace.internal._rand import rand64bits; trace_id = rand64bits(); trace_id_hex = "{:032x}".format(trace_id)' '[("{:032x}-{:016x}".format(trace_id, rand64bits())) for _ in range(100)]'
100000 loops, best of 5: 24.5 usec per loop
```

If instead we already have the hex formatted `trace_id`, we get close to the same timing as we get if we do not include the `trace_id` in `traceparent`:

```
root@b1754db4c2d8:/# python -m timeit --number=100000 --setup='from ddtrace.internal._rand import rand64bits; trace_id = rand64bits(); trace_id_hex = "{:032x}".format(trace_id)' '[("-{:016x}".format(rand64bits())) for _ in range(100)]'
100000 loops, best of 5: 17 usec per loop
root@b1754db4c2d8:/# python -m timeit --number=100000 --setup='from ddtrace.internal._rand import rand64bits; trace_id = rand64bits(); trace_id_hex = "{:032x}".format(trace_id)' '[(trace_id_hex + "-{:016x}".format(rand64bits())) for _ in range(100)]'
100000 loops, best of 5: 18.7 usec per loop
```